arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,66 @@
 #include <Smartcar.h>
 
-void setup() {
-  // put your setup code here, to run once:
+ArduinoRuntime arduinoRuntime;
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+const int TRIGGER_PIN           = 6; // D6
+const int ECHO_PIN              = 7; // D7
+const unsigned int MAX_DISTANCE = 300;
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+SimpleCar car(control);
+int magnitude = 0;                              //This we will need later
+
+void setup()
+{
+    Serial.begin(9600);
+    Serial.setTimeout(200);
+}
+
+void loop()
+{
+  handleInput();
+  delay(35);
+}
+
+void handleInput()
+{
+  float distance = front.getDistance();
+  Serial.println(distance);
+  if (Serial.available())                           //If the user enters input in the serial The car must move according to that input.
+  {
+   String input = Serial.readStringUntil('\n');
+   if (input.startsWith("m"))     
+   {
+     int cSpeed = input.substring(1).toInt();
+     magnitude = cSpeed;                           //We save the user's input in here in order to have it outside of the if scope.                     
+     car.setSpeed(cSpeed); 
+   } else if (input.startsWith("t"))
+   {
+     int cAngle = input.substring(1).toInt();
+     car.setAngle(cAngle);
+     delay(600);                                                      //This delay is needed for the car to turn in a short while and then go back to its straight direction,  
+   }                                                                  // because we dont want the car to to turn around itself for no reason!
+   if (distance > 1 && distance < 200)          //This is the obstacle treshhold. The car must detect obstacle closer than 200
+   {                                                                  // and avoid them
+     handleObstacle();
+   }  
```

Comment:
This doesn't work as expected if the sensors return `0` which is their way of saying "no obstacle detected".